In [3]:
import numpy as np

#
**Solution**
***

##
**Task 1**
***

In [4]:
#Defining eta and probability
etas = np.array([-np.sqrt(3), 0, np.sqrt(3)])
probability = np.array([1/6, 2/3, 1/6])

In [5]:
#Defining variables
beta = 0.9
gamma = 4.0
sigma = 0.16
mu = 0.04
w = 1.0
R_f = 1.0

In [6]:
#Calculating risky return
R  = R_f + mu + sigma * etas #As etas is an array, NumPy automatically calculates for every value
print(f"Rate of return on the risky asset is {R[0]:.2f}, {R[1]:.2f} or {R[2]:.2f}.")

Rate of return on the risky asset is 0.76, 1.04 or 1.32.


In [7]:
#Calculating mean expected rate of return
E_r = np.sum(probability * R)
print(f"The wheigted average rate of return on a risky asset is {E_r}.")

The wheigted average rate of return on a risky asset is 1.04.


In [8]:
#Calculating deviation and variance
deviation = R - E_r

quad = deviation ** 2

In [9]:
var = np.sum(quad * probability)
print(f"The variance is {var}, which is equal to 0.16^2")

The variance is 0.0256, which is equal to 0.16^2


As we can see, both the variance and the expected excess return are equal to the ones given in the table above.

The variance $\sigma$ is given as $0.16^2$, which is equal to $0.0256$\
The expected excess return $\mu$ is given as $0.04$, equal to our calculations of $\mu = 1.04$.

##
**Task 2**
***